app/models/host/managed.rb

```diff
@@ -648,6 +650,10 @@ def associate!(cr, vm)
     self.save!(:validate => false) # don't want to trigger callbacks
   end
 
+  def can_disassociate?
```

Comment:
It is, I was confused with the `compute?` in `managed.rb` which is actually only for unattended.